src/CourseDashboard.cpp

```diff
@@ -1,84 +1,36 @@
 #include "CourseDashboard.hpp"
-#include <stdexcept>
-#include <iostream>
-#include <algorithm>
+#include <fstream>
 
 CourseDashboard::CourseDashboard() {}
 
-void CourseDashboard::showAll()
+bool CourseDashboard::savingToFile(User & c)
 {
-    for (auto & user : users_)
+    std::ofstream out_file{ "../ParticipantsDetails.txt", std::ios::app };
+    if (!out_file)
     {
-        std::cout << user.getAllInfo();
+        std::cerr << "Error" << std::endl;
+        return false;
     }
+        out_file << c;
+        out_file.close();
```

Comment:
you don't need to call close. fstream implements RAII, it closes automatically in destructor.